### PR TITLE
Upgrade WebRender to e491e1ae637b2eed1e7195855d88357e5eb3ddf9

### PR DESCRIPTION
**Pull request by mrobinson**
This change upgrades WebRender to the upstream version as of May 25, 2021.
Upgrading to even newer versions requires even more adaptation of Servo to
adapt to APIs removed upstream.

I have made one concession in this upgrade which is that I have backported the
hit testing APIs that Servo relies on. In general, these are low impact
WebRender changes and this gives us margin to no longer rely on them.

In addition to the changes required in Servo, some tests start to fail. These
are likely due to timing and pixel alignment changes introduced in WebRender.
It is difficult to observe these failures locally and definitely not when using
Servo interactively. The upgrade also exposes some bad behavior that we have
around sticky frames. Hopefully the implementation of sticky in the new layout
system will fix this.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes

### CODE REVIEW

**Comment by mrobinson** [comment]
@bors-servo try

Ah, there is one more thing I forgot to mention about these changes. The adaptation of Servo to the new API includes a bit more conversion to using ClipChains and ClipChainId consistently rather than CilpId. This isn't strictly necessary, but it will be necessary to deal with future version of WebRender.

**Comment by mrobinson** [comment]
This is a draft because it depends on #30321 and also because I'd like to push it through the CI a bit.

**Comment by mrobinson** [comment]
Okay. This one is ready for review.

File: components/layout_thread/lib.rs [hunk 1]
```diff
@@ -109,7 +109,7 @@ use style::traversal_flags::TraversalFlags;
 use style_traits::CSSPixel;
 use style_traits::DevicePixel;
 use style_traits::SpeculativePainter;
-use webrender_api::{units, ColorF, HitTestFlags, ScrollClamping};
+use webrender_api::{units, ColorF, HitTestFlags /*, HitTestFlags*/};
```
Comment (mukilan) [inline components/layout_thread/lib.rs:112]: Can the commented item be removed?

Comment (mrobinson) [reply components/layout_thread/lib.rs]: Thanks for pointing this out. I have removed the commented out use statement.

**Review by mukilan — APPROVED** [review]

**Pull request merged**